include/aws/common/array_list.h

```diff
@@ -61,7 +62,7 @@ int aws_array_list_init_dynamic(
  * by this API. Once this list is full, new items will be rejected.
  */
 AWS_STATIC_IMPL
-void aws_array_list_init_static(
+int aws_array_list_init_static(
```

Comment:
This is a big change, static inits are never expected to fail. If this is what we decide to do, we should go through every call to this function and add the error check.

Reply:
For the static init it's probably reasonable to skip the overflow check. After all, the caller is promising to already have this memory allocated; the overflow check should have happened at that point (and usually static is used for arrays of statically defined size).

Reply:
In that case should we remove the error return and instead assert that the size is reasonable? If the size is incorrect, this would imply programmer error, making an assert the correct way to raise the issue.

Reply:
there is an assert in https://github.com/awslabs/aws-c-common/pull/249/files#diff-a1ef357fc1b5c512b3842ba2d64ffac7R61 
Would it make sense to make this an assert as well, and return to the void function contract?

Reply:
I meant the total size, the one we're checking for overflow. I would rather see that become an assert and the function return void, especially if the caller isn't expected to handle the error.